sys/net/application_layer/nanocoap/sock.c

```diff
@@ -463,6 +467,120 @@ ssize_t nanocoap_sock_get(nanocoap_sock_t *sock, const char *path, void *buf, si
     return _sock_get(sock, path, COAP_TYPE_CON, buf, len);
 }
 
+#ifdef MODULE_NANOCOAP_SOCK_OBSERVE
+static void _async_udp_handler(sock_udp_t *sock, sock_async_flags_t type, void *arg)
+{
+    if (!(type & SOCK_ASYNC_MSG_RECV)) {
+        return;
+    }
+
+    coap_pkt_t pkt;
+    void *payload, *ctx = NULL;
+    coap_observe_client_t *obs = arg;
+    ssize_t res = sock_udp_recv_buf(sock, &payload, &ctx, 0, NULL);
+    if (res <= 0) {
+        return;
+    }
+
+    /* parse response */
+    if (coap_parse(&pkt, payload, res) < 0) {
+        DEBUG("nanocoap: error parsing packet\n");
+        goto out;
+    }
+
+    DEBUG("nanocoap: response code=%i\n", coap_get_code_decimal(&pkt));
+    switch (coap_get_type(&pkt)) {
+    case COAP_TYPE_RST:
+        puts("got RST");
```

Comment:
```suggestion
        DEBUG("nanocoap_sock_observe: ignoring stray RST");
```

Form a protocol point of view we should never receive a RST here. But it certainly could be that this a stray RST from a previous exchange before the other endpoint lost state that just happened to reach us way way later or so.